File: frameworks/runtime-src/Framework/base/MWException.cpp
#include "MWException.h"
#include "cocos2d.h"
#include <new>

using namespace cocos2d;
using namespace std;

MW_FRAMEWORK_BEGIN

MWException *MWException::create(int errCode, const char *msg)
{
    auto pEx = new (nothrow) MWException(errCode, msg);
    if (pEx) {
        pEx->autorelease();
        return pEx;
    }
    return nullptr;
}

MWException::MWException(int errCode, const char *msg)
: _errorCode(errCode)
, _errorMsg(msg)
{
    
}

const char *MWException::what() const
{
    char buffer[1024];
    sprintf(buffer, "Error message: %s\tError code: %d", _errorMsg.c_str(), _errorCode);
    string msg = buffer;
    return msg.c_str();
}

MW_FRAMEWORK_END

File: frameworks/runtime-src/Framework/lua/auto/api/MWException.lua

--------------------------------
-- @module MWException
-- @extend MWObject
-- @parent_module mw

--------------------------------
-- Get error code.<br>
-- return Error code of the exception.
-- @function [parent=#MWException] errorCode 
-- @param self
-- @return int#int ret (return value: int)
        
--------------------------------
-- 
-- @function [parent=#MWException] what 
-- @param self
-- @return char#char ret (return value: char)
        
--------------------------------
-- Create MWException object.<br>
-- param errCode Error code.<br>
-- param msg Error message.<br>
-- return MWException instance which is autoreleased.
-- @function [parent=#MWException] create 
-- @param self
-- @param #int errCode
-- @param #char msg
-- @return MWException#MWException ret (return value: mw.MWException)
        
--------------------------------
-- MWException constructor.<br>
-- param errCode Error code.<br>
-- param msg Error message.
-- @function [parent=#MWException] MWException 
-- @param self
-- @param #int errCode
-- @param #char msg
        
return nil
